src/test/clojure/clara/test_rules.clj

```diff
@@ -2361,10 +2361,31 @@
                   (query check-exception))
               (is false "Scope binding test is expected to throw an exception.")
               (catch Exception e e))]
-
       (is (= {:?w 10, :?t nil}
              (-> e (ex-data) :bindings))))))
 
+(deftest test-include-details-when-exception-is-thrown-in-test-filter
+  (testing "when a condition exception is thrown ensure it contains the necessary details and text"
+    (let [check-exception (assoc (dsl/parse-query [] [[WindSpeed (= ?w windspeed)]
+                                                      [Temperature (= ?t temperature)]
+                                                      [:test (> ?t ?w)]])
+                                 :name "my-test-query")]
+
+      (let [e (try
+                (-> (mk-session [check-exception])
+                    (insert (->WindSpeed 10 "MCI") (->Temperature nil "MCI"))
+                    (fire-rules)
+                    (query check-exception))
+                (is false "Scope binding test is expected to throw an exception.")
+                (catch Exception e e))]
+        (is (= {:bindings {:?w 10, :?t nil}
+                :fact nil
+                :env {}
+                :conditions-and-rules {[:test '(> ?t ?w)] #{[:query "my-test-query"]}}}
+               (-> e (ex-data))))
```

Comment:
It's not too difficult to make assert-ex-data also be able to assert the message, as an additional optional parameter, I'm going to make that change and submit it, it will make it easier to test both message and ex-data of nested exceptions in the future.

Reply:
@WilliamParker i've made a small change to `assert-ex-data` to allow checking the message along with the data as well, and updated the existing test to use the new `(assert-ex-data "message..." {...ex-data...} form)`.

Reply:
+1 to those changes.